sonaranalyzer-dotnet/rspec/cs/S2245_c#.html

```diff
@@ -0,0 +1,38 @@
+<p>When software generates predictable values in a context requiring unpredictability, it may be possible for an attacker to guess the next value that
+will be generated, and use this guess to impersonate another user or access sensitive information.</p>
+<p>As the <code>System.Random</code> class relies on a pseudorandom number generator, it should not be used for security-critical applications or for
+protecting sensitive data. In such context, the <code>System.Cryptography.RandomNumberGenerator</code> class which relies on a cryptographically
+strong random number generator (RNG) should be used in place.</p>
+<h2>Noncompliant Code Example</h2>
+<pre>
+var random = new Random(); // Noncompliant
+byte[] data = new byte[16];
+random.NextBytes(data);
+return BitConverter.ToString(data);
+</pre>
+<h2>Compliant Solution</h2>
+<pre>
+using System.Security.Cryptography;
+...
+var randomGenerator = RandomNumberGenerator.Create(); // Compliant
+byte[] data = new byte[16];
+randomGenerator.GetBytes(data);
+return BitConverter.ToString(data);
+</pre>
+<h2>See</h2>
+<ul>
+  <li> <a href="http://cwe.mitre.org/data/definitions/338.html">MITRE, CWE-338</a> - Use of Cryptographically Weak Pseudo-Random Number Generator
+  (PRNG) </li>
+  <li> <a href="http://cwe.mitre.org/data/definitions/330.html">MITRE, CWE-330</a> - Use of Insufficiently Random Values </li>
+  <li> <a href="http://cwe.mitre.org/data/definitions/326.html">MITRE, CWE-326</a> - Inadequate Encryption Strength </li>
+  <li> <a href="http://cwe.mitre.org/data/definitions/310">MITRE, CWE-310</a> - Cryptographic Issues </li>
+  <li> <a href="https://www.securecoding.cert.org/confluence/x/mAFqAQ">CERT, MSC02-J.</a> - Generate strong random numbers </li>
+  <li> <a href="https://www.securecoding.cert.org/confluence/x/qw4">CERT, MSC30-C.</a> - Do not use the rand() function for generating pseudorandom
+  numbers </li>
+  <li> <a href="https://www.securecoding.cert.org/confluence/x/WYIyAQ">CERT, MSC50-CPP.</a> - Do not use std::rand() for generating pseudorandom
+  numbers </li>
+  <li> OWASP Top 10 2017 Category A3 - Sensitive Data Exposure </li>
+  <li> Derived from FindSecBugs rule <a href="http://h3xstream.github.io/find-sec-bugs/bugs.htm#PREDICTABLE_RANDOM">Predictable Pseudo Random Number
```

Comment:
Does it make sense to talk about `FindSecBugs` in the C# context?

Reply:
Why not? This seems to be the description of the issue